src/mame/machine/nb1414m4.cpp

```diff
@@ -91,8 +97,12 @@ void nb1414m4_device::fill(uint16_t dst, uint8_t tile, uint8_t pal, uint8_t *vra
 
 void nb1414m4_device::insert_coin_msg(uint8_t *vram)
 {
+	if (m_in_game)
+		return;
+
 	int credit_count = (vram[0xf] & 0xff);
-	uint8_t fl_cond = screen().frame_number() & 0x10; /* for insert coin "flickering" */
+	uint8_t fl_cond = (m_frame_count & 0x10) == 0; /* for insert coin "flickering" */
+	m_frame_count++;
```

Comment:
Sounds unlikely that game has own internal frame counter, it's in my humble opinion more possible that the machine_config (again) has naive screen parameters hooked up that causes wrong timings.

Reply:
I guessed from "kozure" and "ninjemak" blinking timing, and when the "INSERT COIN" display message was called, I thought it was counting using the internal work memory of the MCU.
At least, when use frame counter since starting emulation, blinking started was not the same of PCBs.
And, can you tell me what screen parameters of machine_config is?

Reply:
This counter is only for blinking "INSERT COIN", "ONE PLAYER ONLY" and "ONE OR TWO PLAYERS". Should I change the variable name?

Reply:
To make it easier to understand, I changed the name of m_frame_count to m_flickering_count.

Reply:
```
void bigfghtr_state::bigfghtr(machine_config &config)
{
        [...]
	/* video hardware */
	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
	screen.set_raw(XTAL(16'000'000)/2,531,12*8,(64-12)*8, 254, 1*8, 31*8); // guess, matches 59.3 Hz from reference - measured at 59.1358Hz
```
This is the better implementation of CRTC parameters, inputs are `pixel_clock, htotal, hdisplay_start, hdisplay_end, vtotal, vdisplay_start, vdisplay_end`. 
```
void armedf_state::cclimbr2(machine_config &config)
{
       [...]
	/* video hardware */
	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
	screen.set_refresh_hz(60);
	screen.set_vblank_time(ATTOSECONDS_IN_USEC(2500) /* not accurate */);
	screen.set_size(64*8, 32*8);
	screen.set_visarea(14*8, (64-14)*8-1, 2*8, 30*8-1 );
```
This instead is the legacy format. It doesn't really honor the screen timing at all and if SW relies on somehow tight vblank/vdisplay timings it causes all kinds of video and even input lag artifacts. When a driver sports this it is really a cargo cult caused by lack of info about h/vsync timings and imho should be marked as deprecated function off the bat.

So, in order to fix this for _armed.cpp_ you can sub-routine the screen creation with the display resolution as parameters (because I truly guess all of these Nichibutsu games shares same CRTC discrete device). 
For deriving the screen timings the canonical approximation is:
```
htotal = pixel_clock / hsync
vtotal = hsync / vsync
```
Scaled to Hz unit and rounded up to next integer. pixel_clock must be derived from available xtals from PCB, deriving the pixel_clock _divider_ is a matter of common sense (i.e. htotal mustn't be smaller than hdisplay and it is unlikely that a xtal divider isn't a multiple of 2).
For armedf.cpp this translates to:
```
// pixel_clock cannot be /4 because 265,36 falls into the hdisplay area. Could be /3 but 353,82 hotal sounds too short.
htotal = (16/2 MHz) / (15,0735 kHz) = 8000000 / 15073,5 = 530,73 = 531
vtotal = (15,0735 kHz) / (59,1358 Hz) = 15073,5 / 59,1358 = 254,89 = 255
```

A good reference about how analog CRT actually works out is [here](http://datagenetics.com/blog/april12018/index.html), notice that h/vtotals are actually a collection of blanking/sync/overscan events that MAME doesn't currently take into account.

Reply:
Thank you for telling me so much in detail.
I understood what didn't know.

I thought that this blinking by NB1414M4 was expressed by writing a blank in VRAM, just like when displaying characters. The one of reason is that the display command is called every frame while it needs to blink.
And, all blinks that do not pass through the NB1414M4 appear to be represented by a VRAM rewrite by the software. (For example, "CONTINUE PLAY?" in kozure. The blinking timing of "1-UP" also controlled by software.)
For these reason, this hardware seems does not have a blinking function, and it seems to be an operation that simply writes characters or blanks alternately into VRAM every time it is called 16 times.

This was the reason for using the counter, not the screen hardware timing.
Do you think there is a possibility?

Reply:
Another reason. At least, “INSERT COIN” blinking timing after power-on is completely the same between kozure and youma. This starts at the first of 32 frame period (strictly the second frame).
As in the current code, if you count only when the command is called, timing will match exactly.

Comparison video is here :
https://imgur.com/a/i3FjjoB